### PR TITLE
concurrency / it-academy.by / JC1-117-22 / Aliaksei Kharko / Done tasks and demo

### PR DESCRIPTION
**Pull request by KharkoAO**
## Pull request state

Actual status for PR:
- [x] implemented
- [x] reviewed
- [ ] fixed
- [ ] approved

### CODE REVIEW

File: src/main/java/com/rakovets/course/java/core/practice/concurrency/producer_queue_consumer/exception/UserInputException.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package com.rakovets.course.java.core.practice.concurrency.producer_queue_consumer.exception;
+
+public class UserInputException extends Exception{
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/concurrency/producer_queue_consumer/exception/UserInputException.java:3]:
```suggestion
public class UserInputException extends Exception {
```

File: src/main/java/com/rakovets/course/java/core/practice/concurrency/producer_queue_consumer/Report.txt [hunk 2]
```diff
@@ -0,0 +1,69 @@
+2022-11-09 01:17:21.091 - Consumer 3 - I slept 2 seconds
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/concurrency/producer_queue_consumer/Report.txt:1]: remove it

File: src/main/java/com/rakovets/course/java/core/practice/concurrency/producer_queue_consumer/Producer.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    @Override
+    public void run() {
+        Scanner scanner = new Scanner(System.in);
+        String input = "";
+
+        while (!input.equals("-1")) {
+            try {
+                input = scanner.nextLine();
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/concurrency/producer_queue_consumer/Producer.java:25]: use `nextInt()`

File: src/main/java/com/rakovets/course/java/core/practice/concurrency/producer_queue_consumer/Consumer.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    public Consumer(String name, QueueNumbersFromStream queue) {
+        super(name);
+        this.queue = queue;
+    }
+
+    @Override
+    public void run() {
+        try {
+            BufferedWriter writer = new BufferedWriter(
+                    new FileWriter("D://JavaProjects/course-java-core" +
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/concurrency/producer_queue_consumer/Consumer.java:23]: hardcode

File: src/main/java/com/rakovets/course/java/core/practice/concurrency/producer_queue_consumer/Consumer.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+                } else {
+                    writer.write(timestamp +
+                            " - " + Thread.currentThread().getName() +
+                            " - " + "...");
+                    writer.newLine();
+                    writer.flush();
+                    Thread.sleep(MILLS_IN_SECOND);
+                }
+            }
+        } catch (IOException | InterruptedException | NullPointerException e) {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/concurrency/producer_queue_consumer/Consumer.java:44]: NPE?

File: src/main/java/com/rakovets/course/java/core/practice/concurrency/producer_queue_consumer/Consumer.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.sql.Timestamp;
+
+public class Consumer extends Thread {
+    private Timestamp timestamp = new Timestamp(System.currentTimeMillis());
+    private QueueNumbersFromStream queue;
+
+    private final int MILLS_IN_SECOND = 1000;
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/concurrency/producer_queue_consumer/Consumer.java:12]:
```suggestion
    private final int MILLS_IN_SECOND = 1000;
    
    private Timestamp timestamp = new Timestamp(System.currentTimeMillis());
    private QueueNumbersFromStream queue;

```

**Review by rakovets — CHANGES_REQUESTED** [review]
need to fix and add solution for Backup utility

File: src/main/java/com/rakovets/course/java/core/practice/concurrency/producer_queue_consumer/exception/UserInputException.java (hunk 1 above)
Comment (KharkoAO) [reply src/main/java/com/rakovets/course/java/core/practice/concurrency/producer_queue_consumer/exception/UserInputException.java]: this fixed

File: src/main/java/com/rakovets/course/java/core/practice/concurrency/producer_queue_consumer/Report.txt (hunk 2 above)
Comment (KharkoAO) [reply src/main/java/com/rakovets/course/java/core/practice/concurrency/producer_queue_consumer/Report.txt]: this fixed

File: src/main/java/com/rakovets/course/java/core/practice/concurrency/producer_queue_consumer/Producer.java (hunk 3 above)
Comment (KharkoAO) [reply src/main/java/com/rakovets/course/java/core/practice/concurrency/producer_queue_consumer/Producer.java]: this fixed

File: src/main/java/com/rakovets/course/java/core/practice/concurrency/producer_queue_consumer/Consumer.java (hunk 4 above)
Comment (KharkoAO) [reply src/main/java/com/rakovets/course/java/core/practice/concurrency/producer_queue_consumer/Consumer.java]: this fixed

File: src/main/java/com/rakovets/course/java/core/practice/concurrency/producer_queue_consumer/Consumer.java (hunk 5 above)
Comment (KharkoAO) [reply src/main/java/com/rakovets/course/java/core/practice/concurrency/producer_queue_consumer/Consumer.java]: this fixed

File: src/main/java/com/rakovets/course/java/core/practice/concurrency/producer_queue_consumer/Consumer.java (hunk 6 above)
Comment (KharkoAO) [reply src/main/java/com/rakovets/course/java/core/practice/concurrency/producer_queue_consumer/Consumer.java]: this fixed

**Pull request closed without merging**